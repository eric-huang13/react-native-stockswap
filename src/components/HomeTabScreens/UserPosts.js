import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  Image,
  TouchableOpacity,
  SafeAreaView,
  Modal,
} from 'react-native';
import LikeInactiveIcon from '../../icons/LikeInactiveIcon';
import CommentIcon from '../../icons/CommentIcon';
import ReportModal from '../HomeTabComponents/ReportModal';
import ShareToModal from '../HomeTabComponents/ShareToModal';
import {moderateScale, scale} from '../../util/responsiveFont';

export default class UserPosts extends Component {
  constructor(props) {
    super(props);
    this.state = {
      shouldShow: false,
      reportModalState: false,
      shareModalState: false,
    };
  }
  accountId = this.props.userAccount.id;

  navigationByCondition = (post) => {
    const {navigation} = this.props;
    if (post.userId === this.accountId) {
      navigation.navigate({
        name: 'MyProfile',
        params: {id: post.id},
      });
    } else {
      navigation.navigate({
        name: 'Profile',
        params: {id: post.userId},
      });
    }
  };
  dropDownSelect(post, userAccount) {
    this.setState({shouldShow: false});
    this.props.navigation.navigate({
      name: 'EditPost',
      params: {post, userAccount},
    });
  }
  reportModal(item) {
    this.setState({reportModalState: item, shouldShow: false});
  }
  shareModal(item) {
    this.setState({shareModalState: item, shouldShow: false});
  }
  render() {
    const {shouldShow, reportModalState, shareModalState} = this.state;

    const {post, comments, reply, userAccount} = this.props;

    const filteredComments = comments.filter(
      (comment) => comment.postId === post.id,
    );
    const lastComment = filteredComments[filteredComments.length - 1];
    // console.log(this.props.navigation, 'props in post');
    return (
      <SafeAreaView style={style.container}>
        <Modal
          transparent={true}
          visible={reportModalState}
          animationType="slide">
          <ReportModal reportModal={this.reportModal.bind(this)} />
        </Modal>
        <Modal
          transparent={true}
          visible={shareModalState}
          animationType="slide">
          <ShareToModal shareModal={this.shareModal.bind(this)} />
        </Modal>
        <View style={style.postNameContainer}>
          <TouchableOpacity
            key={post.id}
            onPress={() => this.navigationByCondition(post)}>
            <View style={style.profileImageContainer}>
              <Image
                style={style.postUserImage}
                source={{uri: post.profileImg}}
              />
              <Text style={style.postUserName}>{post.name}</Text>
            </View>
          </TouchableOpacity>
          {userAccount.id === post.userId ? (
            <View style={style.dotsDropdownContainer}>
              <TouchableOpacity
                onPress={() =>
                  this.setState({
                    shouldShow: !shouldShow,
                  })
                }>
                <Text style={style.dotsButton}>...</Text>
              </TouchableOpacity>
              {this.state.shouldShow ? (
                <View style={style.dropdownEdit}>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect(post, userAccount)}>
                    <Text style={style.dropDownText}>Edit post</Text>
                  </TouchableOpacity>
                  <View style={style.dropDownTextReportContainer}>
                    <Text style={style.dropDownText}>Remove post</Text>
                  </View>
                </View>
              ) : null}
            </View>
          ) : (
            <View style={style.dotsDropdownContainer}>
              <TouchableOpacity
                onPress={() =>
                  this.setState({
                    shouldShow: !shouldShow,
                  })
                }>
                <Text style={style.dotsButton}>...</Text>
              </TouchableOpacity>
              {this.state.shouldShow ? (
                <View style={style.dropdown}>
                  <Text style={style.dropDownText}>Repost</Text>
                  <Text style={style.dropDownText}>Copy link</Text>
                  <TouchableOpacity onPress={() => this.shareModal(true)}>
                    <Text style={style.dropDownText}>Share to...</Text>
                  </TouchableOpacity>
                  <Text style={style.dropDownText}>Turn on notifications</Text>
                  <TouchableOpacity
                    key={post.id}
                    onPress={() => this.reportModal(true)}>
                    <View style={style.dropDownTextReportContainer}>
                      <Text style={style.dropDownText}>Report</Text>
                    </View>
                  </TouchableOpacity>
                </View>
              ) : null}
            </View>
          )}
        </View>
        <TouchableOpacity
          onPress={() =>
            this.props.navigation.navigate({
              name: 'PostScreen',
              params: {post, filteredComments, reply, userAccount},
            })
          }>
          <Image
            style={[style.image, shouldShow ? {opacity: 0.2} : null]}
            source={{uri: post.img}}
          />
        </TouchableOpacity>

        <View style={style.detailsContainer}>
          <Text style={style.timestamp}>{post.timestamp}</Text>

          <View style={style.likesContainer}>
            <View style={style.iconContainer}>
              <LikeInactiveIcon />
              <Text style={style.likes}>{post.likes}</Text>
            </View>
            <View style={style.iconContainer}>
              <CommentIcon />
              <Text style={style.comments}>{post.comments}</Text>
            </View>
          </View>
        </View>
        <TouchableOpacity
          onPress={() =>
            this.props.navigation.navigate({
              name: 'PostScreen',
              params: {post, filteredComments, reply, userAccount},
            })
          }>
          {post.body.length < 80 ? (
            <Text style={style.body}>{post.body}</Text>
          ) : (
            <Text style={style.body}>
              {' '}
              {post.body.substring(0, 80)}...
              <Text style={style.more}>{'       '}More</Text>
            </Text>
          )}
        </TouchableOpacity>

        <View style={style.commentContainer}>
          <View style={style.commentContainer}>
            <TouchableOpacity
              onPress={() =>
                this.props.navigation.navigate({
                  name: 'UserCommentList',
                  params: {postId: post.id, userAccount},
                })
              }>
              {/* <View style={style.headerContainer}>
                <Text style={style.allComments}>View all comments</Text>
              </View> */}

              {lastComment ? (
                <View style={style.lastCommentContainer}>
                  <Text style={style.lastCommentName}>{lastComment.name}:</Text>
                  <Text style={style.lastCommentBody}>
                    {lastComment.body.length < 55
                      ? `${lastComment.body}`
                      : `${lastComment.body.substring(0, 55)}...`}
                  </Text>
                </View>
              ) : null}
            </TouchableOpacity>
          </View>
        </View>
      </SafeAreaView>
    );
  }
}

const style = StyleSheet.create({
  container: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    marginTop: moderateScale(3),
    paddingVertical: moderateScale(18),
    paddingHorizontal: moderateScale(10),
    backgroundColor: '#2a334a',
  },
  image: {
    height: scale(184),
    width: '100%',
    borderRadius: moderateScale(10),
  },
  postNameContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: moderateScale(8.5),
    justifyContent: 'space-between',
  },
  profileImageContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  postUserImage: {
    height: moderateScale(40),
    width: moderateScale(40),
    borderRadius: moderateScale(50),
  },
  postUserName: {
    color: '#FFFFFF',
    fontSize: moderateScale(16),
    marginLeft: moderateScale(8),
    fontFamily: 'Montserrat-Bold',
  },
  detailsContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginTop: moderateScale(8),
  },
  likesContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  timestamp: {
    fontSize: moderateScale(12.5),
    color: 'lightgrey',
    fontFamily: 'Montserrat-Italic',
  },
  iconContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  likes: {
    fontSize: moderateScale(16),
    color: 'lightgrey',
    fontFamily: 'Montserrat-Medium',
    marginLeft: moderateScale(3),
    marginRight: moderateScale(14),
  },
  comments: {
    fontSize: moderateScale(16),
    color: 'lightgrey',
    fontFamily: 'Montserrat-Medium',
    marginRight: moderateScale(1),
    marginLeft: moderateScale(3),
  },
  body: {
    fontSize: moderateScale(15),
    color: '#FFFFFF',
    marginTop: moderateScale(10),
    marginBottom: moderateScale(4),
    fontFamily: 'Montserrat-Medium',
  },
  more: {
    fontSize: moderateScale(13),
    color: '#B8A0FF',
    fontFamily: 'Montserrat-SemiBoldItalic',
    //   alignSelf:'flex-end',
    //  textAlign:'right'
  },
  commentContainer: {
    marginTop: moderateScale(4),
  },
  allComments: {
    color: '#8b64ff',
    fontStyle: 'italic',
    fontSize: moderateScale(14),
  },
  lastCommentContainer: {
    // marginTop: 1,
  },
  lastCommentName: {
    color: '#999999',
    fontFamily: 'Montserrat-Bold',
    marginBottom: moderateScale(1),
  },
  lastCommentBody: {
    color: '#FFFFFF',
    fontSize: moderateScale(13),
    fontFamily: 'Montserrat-Regular',
  },
  dotsDropdownContainer: {
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    alignContent: 'center',
  },
  dotsButton: {
    alignSelf: 'flex-end',
    color: 'white',
    fontWeight: 'bold',
    fontSize: moderateScale(20),
    marginBottom: moderateScale(10),
  },
  dropdownEdit: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: moderateScale(1),
    marginBottom: moderateScale(-60),
    backgroundColor: '#2C3957',
    zIndex: 1,
    paddingVertical: moderateScale(6),
    // paddingHorizontal:10,
  },
  dropdown: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: moderateScale(1),
    marginBottom: moderateScale(-135),
    backgroundColor: '#2C3957',
    zIndex: 1,
    paddingVertical: moderateScale(6),
    // paddingHorizontal:10,
  },
  dropDownText: {
    color: 'white',
    fontSize: moderateScale(16),
    marginHorizontal: moderateScale(12),
    fontFamily: 'Montserrat-Medium',
  },
  dropDownTextReportContainer: {
    borderTopWidth: moderateScale(1),
    borderTopColor: '#CBCDD7',
    paddingTop: moderateScale(4),
    backgroundColor: '#2C3957',
  },
});
