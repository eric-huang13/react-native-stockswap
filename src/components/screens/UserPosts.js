import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  Image,
  TouchableOpacity,
  SafeAreaView,
} from 'react-native';
import LikeInactiveIcon from '../../icons/LikeInactiveIcon'
import CommentIcon from '../../icons/CommentIcon'

export default class UserPosts extends Component {
  constructor(props) {
    super(props);
    this.state = {
      shouldShow: false,
    };
  }
  navigationByCondition = post => {
    const {navigation} = this.props;
    if (post.userId === this.accountId) {
      navigation.navigate({
        name: 'MyProfile',
        params: {id: post.id},
      })
    } else {
      navigation.navigate({
        name: 'Profile',
        params: {id: post.userId}      })
    }
  };

  render() {
    const {shouldShow} = this.state;

    const {post, comments, reply, userAccount} = this.props;
    const filteredComments = comments.filter(
      (comment) => comment.postId === post.id,
    );
    const lastComment = filteredComments[filteredComments.length - 1];
    // console.log(this.props.navigation, 'props in post');
    return (
      <SafeAreaView style={style.container}>
        <View style={style.postNameContainer}>
        <TouchableOpacity
                    key={post.id}
                    onPress={()=>this.navigationByCondition(post)             
                    }>
          <View style={style.profileImageContainer}>
            <Image
              style={style.postUserImage}
              source={{uri: post.profileImg}}
            />
            <Text style={style.postUserName}>{post.name}</Text>
          </View>
          </TouchableOpacity>

          <View style={style.dotsDropdownContainer}>
            <TouchableOpacity
              onPress={() =>
                this.setState({
                  shouldShow: !shouldShow,
                })
              }>
              <Text style={style.dotsButton}>...</Text>
           
            </TouchableOpacity>
            {this.state.shouldShow ? (
              <View style={style.dropdown}>
                <Text style={style.dropDownText}>Repost</Text>
                <Text style={style.dropDownText}>Copy link</Text>
                <Text style={style.dropDownText}>Turn on notifications</Text>
                <View style={style.dropDownTextReportContainer}>
                  <Text style={style.dropDownText}>Report</Text>
                </View>
              </View>
            ) : null}
          </View>
        </View>
        <TouchableOpacity
          onPress={() =>
            this.props.navigation.navigate({
              name: 'PostScreen',
              params: {post, filteredComments, reply, userAccount},
            })
          }>
          <Image style={style.image} source={{uri: post.img}} />
        </TouchableOpacity>

        <View style={style.detailsContainer}>
          <Text style={style.timestamp}>{post.timestamp}</Text>

          <View style={style.likesContainer}>
              <View style={style.iconContainer}>
                <LikeInactiveIcon/>
              <Text style={style.likes}>{post.likes}</Text>
              </View>
              <View style={style.iconContainer}>
                <CommentIcon/>
              <Text style={style.comments}>{post.comments}</Text>
              </View>
            </View>
        </View>
        <TouchableOpacity
          onPress={() =>
            this.props.navigation.navigate({
              name: 'PostScreen',
              params: {post, filteredComments, reply, userAccount},
            })
          }>
          <Text style={style.body}>
            {' '}
            {post.body.length < 88
              ? `${post.body}`
              : `${post.body.substring(0, 88)}...`}{' '}
            <Text style={style.more}>{'       '}More</Text>
          </Text>
        </TouchableOpacity>

        <View style={style.commentContainer}>
          <View style={style.commentContainer}>
            <TouchableOpacity
              onPress={() =>
                this.props.navigation.navigate({
                  name: 'PostScreen',
                  params: {post, filteredComments, reply, userAccount},
                })
              }>
              {/* <View style={style.headerContainer}>
                <Text style={style.allComments}>View all comments</Text>
              </View> */}

              {lastComment ? (
                <View style={style.lastCommentContainer}>
                  <Text style={style.lastCommentName}>{lastComment.name}:</Text>
                  <Text style={style.lastCommentBody}>
                    {lastComment.body.length < 55
                      ? `${lastComment.body}`
                      : `${lastComment.body.substring(0, 55)}...`}
                  </Text>
                </View>
              ) : null}
            </TouchableOpacity>
          </View>
        </View>
      </SafeAreaView>
    );
  }
}

const style = StyleSheet.create({
  container: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    marginTop: 3,
    paddingVertical: 18,
    paddingHorizontal: 10,
    backgroundColor: '#2a334a',
  },
  image: {
    height: 184,
    width: '100%',
    borderRadius: 10,
  },
  postNameContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 8.5,
    justifyContent: 'space-between',
  },
  profileImageContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  postUserImage: {
    height: 40,
    width: 40,
    borderRadius: 50,
  },
  postUserName: {
    color: '#FFFFFF',
    fontSize: 16,
    marginLeft: 8,
    fontFamily:'Montserrat-Bold',
  },
  detailsContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginTop: 8,
  },
  likesContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  timestamp: {
    fontSize: 12.5,
    color: 'lightgrey',
    fontFamily:'Montserrat-Italic',
  },
  iconContainer:{
    flexDirection:'row',
     alignItems:'center',
     justifyContent:'space-between',
  },
  likes: {
    fontSize: 16,
    color: 'lightgrey',
    fontFamily:'Montserrat-Medium',
    marginLeft:3,
    marginRight:14,
    
  },
  comments: {
    fontSize: 16,
    color: 'lightgrey',
    fontFamily:'Montserrat-Medium',
    marginRight: 1,
    marginLeft:3,
  },
  body: {
    fontSize: 15,
    color: '#FFFFFF',
    marginTop: 10,
    marginBottom: 4,
    fontFamily:'Montserrat-Medium',
  },
  more: {
    fontSize: 13,
    color: '#B8A0FF',
    fontFamily:'Montserrat-SemiBoldItalic',
  },
  commentContainer: {
    marginTop: 4,
  },
  allComments: {
    color: '#8b64ff',
    fontStyle: 'italic',
    fontSize: 14,
  },
  lastCommentContainer: {
    // marginTop: 1,
  },
  lastCommentName: {
    color: '#999999',
    fontFamily:'Montserrat-Bold',
    marginBottom:1,
  },
  lastCommentBody: {
    color: '#FFFFFF',
    fontSize:13,
    fontFamily:'Montserrat-Regular',
  },
  dotsDropdownContainer: {
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    alignContent: 'center',
  },
  dotsButton: {
    alignSelf: 'flex-end',
    color: 'white',
    fontWeight: 'bold',
    fontSize: 20,
  
  },
  dropdown: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: 1,
    marginBottom: -125,
    backgroundColor: '#2C3957',
    zIndex: 1,
    paddingVertical: 6,
    // paddingHorizontal:10,
  },
  dropDownText: {
    color: 'white',
    fontSize: 16,
    marginHorizontal: 12,
    fontFamily:'Montserrat-Medium',
  },
  dropDownTextReportContainer: {
    borderTopWidth: 1,
    borderTopColor: '#CBCDD7',
    paddingTop: 4,
    backgroundColor:'#2C3957'
  },
});
