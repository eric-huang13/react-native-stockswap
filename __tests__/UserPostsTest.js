import React from 'react';
import UserPosts from '../src/components/HomeTabScreens/UserPosts';
import renderer from 'react-test-renderer';


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));

test('renders correctly', () => {
    const tree = renderer.create(<UserPosts userAccount={{id:'test' }} post={{id:'test', body:'testbody' }} comments={[]}/>).toJSON();
  expect(tree).toMatchSnapshot();
});