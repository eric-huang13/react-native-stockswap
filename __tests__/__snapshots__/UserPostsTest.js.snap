// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flexDirection": "column",
      "justifyContent": "space-between",
      "marginTop": 4.237226277372263,
      "paddingHorizontal": 14.124087591240876,
      "paddingVertical": 25.423357664233578,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3e4d6c",
          "borderRadius": 20,
          "elevation": 5,
          "flex": 1,
          "marginTop": 310,
          "padding": 14,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#B8A0FF",
              "fontFamily": "Montserrat-Bold",
              "fontSize": 19.77372262773723,
              "marginBottom": 18,
            }
          }
        >
          Close
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 19.77372262773723,
            "marginHorizontal": 12,
            "paddingVertical": 2,
          }
        }
      >
        Why would you like to report this post?
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
            "height": "50%",
            "justifyContent": "space-between",
            "marginTop": 30,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 16.94890510948905,
              "marginHorizontal": 12,
              "paddingVertical": 2,
            }
          }
        >
          Option 1
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 16.94890510948905,
              "marginHorizontal": 12,
              "paddingVertical": 2,
            }
          }
        >
          Option 2
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 16.94890510948905,
              "marginHorizontal": 12,
              "paddingVertical": 2,
            }
          }
        >
          Option 3
        </Text>
      </View>
    </View>
  </Modal>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3e4d6c",
          "borderRadius": 28.248175182481752,
          "elevation": 5,
          "flex": 1,
          "marginTop": 564.963503649635,
          "padding": 19.77372262773723,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#B8A0FF",
              "fontFamily": "Montserrat-Bold",
              "fontSize": 19.77372262773723,
              "marginBottom": 25.423357664233578,
            }
          }
        >
          Close
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 19.77372262773723,
            "marginHorizontal": 16.94890510948905,
            "paddingVertical": 2.8248175182481754,
          }
        }
      >
        Share to...
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
            "height": "50%",
            "justifyContent": "space-between",
            "marginTop": 42.372262773722625,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 16.94890510948905,
              "marginHorizontal": 16.94890510948905,
              "paddingVertical": 2.8248175182481754,
            }
          }
        >
          Option 1
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 16.94890510948905,
              "marginHorizontal": 16.94890510948905,
              "paddingVertical": 2.8248175182481754,
            }
          }
        >
          Option 2
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 16.94890510948905,
              "marginHorizontal": 16.94890510948905,
              "paddingVertical": 2.8248175182481754,
            }
          }
        >
          Option 3
        </Text>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 12.005474452554743,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "borderRadius": 70.62043795620437,
              "height": 56.496350364963504,
              "width": 56.496350364963504,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-Bold",
              "fontSize": 22.598540145985403,
              "marginLeft": 11.299270072992702,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "flexDirection": "column",
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "white",
              "fontSize": 28.248175182481752,
              "fontWeight": "bold",
              "marginBottom": 14.124087591240876,
            }
          }
        >
          ...
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 14.124087591240876,
          "height": 335.7664233576642,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 11.299270072992702,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "lightgrey",
          "fontFamily": "Montserrat-Italic",
          "fontSize": 17.655109489051092,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={22.598540145985403}
          bbWidth={26.835766423357665}
          focusable={false}
          height={22.598540145985403}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 22,
                "width": 26,
              },
            ]
          }
          vbHeight={16}
          vbWidth={19}
          width={26.835766423357665}
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M9.492 16c-.256 0-.503-.093-.695-.261a208.417 208.417 0 00-2.045-1.76l-.003-.002c-1.812-1.544-3.377-2.877-4.465-4.19C1.067 8.316.5 6.925.5 5.404c0-1.478.507-2.841 1.427-3.84A4.836 4.836 0 015.524 0C6.562 0 7.513.328 8.35.975c.422.327.805.727 1.142 1.193.337-.466.72-.866 1.143-1.193A4.524 4.524 0 0113.46 0c1.389 0 2.667.556 3.597 1.566.92.998 1.427 2.361 1.427 3.839 0 1.52-.567 2.913-1.784 4.381-1.088 1.314-2.653 2.647-4.464 4.19-.62.528-1.32 1.126-2.049 1.763-.192.168-.44.261-.695.261zM5.524 1.053A3.794 3.794 0 002.701 2.28c-.74.803-1.148 1.912-1.148 3.125 0 1.279.476 2.423 1.542 3.71 1.03 1.242 2.562 2.548 4.337 4.06l.003.003c.62.529 1.324 1.128 2.055 1.768a245.89 245.89 0 012.063-1.77c1.774-1.513 3.306-2.819 4.336-4.062 1.066-1.286 1.542-2.43 1.542-3.71 0-1.212-.408-2.321-1.148-3.124a3.794 3.794 0 00-2.823-1.227c-.8 0-1.533.255-2.181.756-.578.446-.98 1.011-1.216 1.406a.66.66 0 01-.571.325.66.66 0 01-.571-.325c-.236-.395-.638-.96-1.216-1.406a3.494 3.494 0 00-2.181-.756z"
              fill={4294967295}
              fillRule={1}
              propList={
                Array [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          style={
            Object {
              "color": "lightgrey",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 22.598540145985403,
              "marginLeft": 4.237226277372263,
              "marginRight": 19.77372262773723,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={22.598540145985403}
          bbWidth={24.010948905109487}
          focusable={false}
          height={22.598540145985403}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 22,
                "width": 24,
              },
            ]
          }
          vbHeight={16}
          vbWidth={17}
          width={24.010948905109487}
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M14.776 0H2.224C1.274 0 .5.773.5 1.724v9.195c0 .95.773 1.724 1.724 1.724h3.978l1.317 2.635c.23.459.587.722.981.722s.751-.263.98-.722l1.318-2.635h3.978c.95 0 1.724-.773 1.724-1.724V1.724C16.5.774 15.727 0 14.776 0zm.787 10.919a.787.787 0 01-.787.786h-4.268c-.177 0-.34.1-.419.26l-1.447 2.893a.57.57 0 01-.142.196.57.57 0 01-.142-.196L6.91 11.964a.469.469 0 00-.42-.259H2.225a.787.787 0 01-.787-.786V1.724c0-.434.353-.786.787-.786h12.552c.434 0 .787.352.787.786v9.195z"
              fill={4294967295}
              fillRule={1}
              propList={
                Array [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          style={
            Object {
              "color": "lightgrey",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 22.598540145985403,
              "marginLeft": 4.237226277372263,
              "marginRight": 1.4124087591240877,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 21.186131386861312,
          "marginBottom": 5.649635036496351,
          "marginTop": 14.124087591240876,
        }
      }
    >
      testbody
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 5.649635036496351,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 5.649635036496351,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
